kolibri/core/webpack/hooks.py

```diff
@@ -0,0 +1,291 @@
+"""
+Kolibri Webpack hooks
+---------------------
+
+To manage assets, we use the webpack format. In order to have assets bundled in,
+you should put them in ``yourapp/assets/src``.
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import json
+import logging
+import os
+import time
+
+from django.conf import settings as django_settings
+from django.contrib.staticfiles.storage import staticfiles_storage
+from django.utils.functional import cached_property
+from django.utils.safestring import mark_safe
+
+from kolibri.core.webpack.utils import render_as_url
+from kolibri.plugins import hooks
+
+from . import settings
+
+
+class BundleNotFound(Exception):
+    pass
+
+class WebpackError(EnvironmentError):
+    pass
+
+
+logger = logging.getLogger(__name__)
+
+
+class WebpackBundleHook(hooks.KolibriHook):
+    """
+    This is the abstract hook class that all plugins that wish to load any
+    assets into the front end must implement, in order for them to be part of
+    the webpack asset loading pipeline.
+    """
+
+    # : You should set a unique human readable name
+    unique_slug = ""
+
+    # : Relative path to js source file for webpack to use as entry point
+    # : For instance: "kolibri/core/assets/src/kolibri_core_app.js"
+    src_file = ""
+
+    # : The static directory where you want stuff to be written to
+    static_dir = "kolibri/core/static"
+
+    def __init__(self, *args, **kwargs):
+        super(WebpackBundleHook, self).__init__(*args, **kwargs)
+
+        # Verify the uniqueness of the slug
+        # It can be '0' in the parent class constructor
+        assert \
+            len([x.unique_slug == self.unique_slug for x in self.registered_hooks]) <= 1, \
```

Comment:
Oh, and IMO it would be *nice* if a conflicting definition raised the error in the snippet below when the module containing the classes is loaded.

```python
class Foo(WebpackBundleHook):
  slug = "foo"

class Bar(WebpackBundleHook):
  slug = "foo"
```

But as you say it's an implementation detail of WebpackBundleHook and not really a feature of the hook system. So I guess I'm wondering if you think it makes sense to *make* this kind of thing a feature of the hook system?